### ISSUE TITLE
IP address field

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/quickwit#1381]
IP address field support!

Ideally it should have some flavor of trie indexing  and be plugged to range queries (See also #939)

**Comment by kstaken** [comment]
Both IPv4 and IPv6 can be handled by storing the address as u128. Ideally the query syntax would support both exact matches and range matches based on both explicit ranges and CIDR notation.

```
ip_field:129.0.5.120 - exact match
ip_field:129.9.3.0/24 = ip_field:[129.9.3.0 TO 129.9.3.255] = ip_field:[2164851456 TO 2164851711]
```

**Comment by fulmicoton** [comment]
Splunk, ES, oracle also support cidrmatch in their query language. The above query then looks as follows:
 `ip_field:129.9.3.0/24`.

Lucene uses a single 128bits scalar to represent ip addresses.
IpV4 get a special prefix. DocValues compression is probably not great but well, at least it is nicer than the equivalent fast field codec.

https://github.com/apache/lucene/blob/d5d6dc079395c47cd6d12dcce3bcfdd2c7d9dc63/lucene/core/src/java/org/apache/lucene/document/InetAddressPoint.java

**Comment by guilload** [comment]
Closed via #2181.
